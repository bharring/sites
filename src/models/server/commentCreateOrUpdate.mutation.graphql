mutation CommentCreateOrUpdate($input: CommentInput) {
  commentCreateOrUpdate(input: $input) {
    comment {
      id
      entity_type
      entity_id
      body
      created_at
      created_by {
        id
        tenantId
        email
        phone
        firstName
        lastName
        organizations {
          id
          name
          subdomain
          roles
        }
      }
    }
  }
}
