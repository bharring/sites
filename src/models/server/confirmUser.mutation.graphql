mutation ConfirmUser($userConfirmationInput: UserConfirmationInput) {
  userManagement(userConfirmationInput: $userConfirmationInput) {
    ... on UserConfirmationResultSuccess {
      __typename
      userConfirmation {
        code
        username
      }
    }
    ... on CodeMismatchError {
      username
      code
      message
    }
    ... on UserNotFoundError {
      message
      username
    }
    ... on NotAuthorizedException {
      message
      username
    }
  }
}
