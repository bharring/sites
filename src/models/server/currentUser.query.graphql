query CurrentUser {
  currentUser {
    email
    firstName
    id
    lastName
    phone
    tenantId
    organizations {
      id
      name
      roles
      subdomain
    }
  }
}
