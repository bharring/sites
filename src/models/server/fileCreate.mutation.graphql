mutation FileCreate($input: FileCreateInput!) {
  fileCreate(input: $input) {
    type
    name
    description
    url
    metaFields {
      name
      value
    }
  }
}
