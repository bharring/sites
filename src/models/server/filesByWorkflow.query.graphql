query FilesByWorkflow($workflowId: String!) {
  filesByWorkflow(workflowId: $workflowId) {
    id
    key
    type
    name
    workflowId
    stepId
    contractName
    url
  }
}
