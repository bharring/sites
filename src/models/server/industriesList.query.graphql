query IndustriesList {
  industriesList {
    items {
      ... on Industry {
        id
        name
        description
      }
    }
    nextToken
    scannedCount
  }
}
