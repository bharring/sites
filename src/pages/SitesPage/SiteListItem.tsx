import { IonLabel } from '@ionic/react';

import { Site } from '../../models/server/schema';

export const SiteListItem: React.FC<{ site: Site | null }> = ({ site }) => (
  <IonLabel>
    {site?.name && <h2>{site.name}</h2>}
    {site?.address && (
      <h3>{`${site.address.street || ''}, ${site.address.locality || ''}, ${site.address.region || ''} ${
        site.address.postalCode || ''
      }`}</h3>
    )}
    {/* TODO: <p>{ site | formatDistance:currentLocation }</p> */}
  </IonLabel>
);

export default SiteListItem;
