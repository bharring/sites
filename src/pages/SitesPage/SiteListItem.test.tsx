import '@testing-library/jest-dom';

import renderer from 'react-test-renderer';

import { Site } from '../../models/server/schema';
import SiteListItem from './SiteListItem';

describe('<SiteListItem />', () => {
  test('renders snapshot full address', () => {
    const site: Site = {
      name: 'Site Name',
      address: {
        street: 'street',
        locality: 'locality',
        region: 'region',
        postalCode: 'postalCode',
      },
    };
    const tree = renderer.create(<SiteListItem site={site} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders snapshot missing address', () => {
    const site: Site = { name: 'Site Name' };
    const tree = renderer.create(<SiteListItem site={site} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders snapshot missing properties', () => {
    const site: Site = {};
    const tree = renderer.create(<SiteListItem site={site} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
