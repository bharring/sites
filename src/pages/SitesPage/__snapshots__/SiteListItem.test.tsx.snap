// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiteListItem /> renders snapshot full address 1`] = `
<ion-label>
  <h2>
    Site Name
  </h2>
  <h3>
    street, locality, region postalCode
  </h3>
</ion-label>
`;

exports[`<SiteListItem /> renders snapshot missing address 1`] = `
<ion-label>
  <h2>
    Site Name
  </h2>
</ion-label>
`;

exports[`<SiteListItem /> renders snapshot missing properties 1`] = `<ion-label />`;
